prov/efa/src/rxr/rxr_ep.c

```diff
@@ -1596,12 +1597,30 @@ void rxr_ep_progress_internal(struct rxr_ep *ep)
 	dlist_foreach_container_safe(&ep->tx_entry_queued_list,
 				     struct rxr_tx_entry,
 				     tx_entry, queued_entry, tmp) {
-		if (tx_entry->state == RXR_TX_QUEUED_CTRL)
-			ret = rxr_pkt_post_ctrl(ep, RXR_TX_ENTRY, tx_entry,
-						tx_entry->queued_ctrl.type,
-						tx_entry->queued_ctrl.inject);
-		else
+		if (tx_entry->state == RXR_TX_QUEUED_CTRL) {
+			peer = rxr_ep_get_peer(ep, tx_entry->addr);
+			if (tx_entry->queued_ctrl.type >= RXR_DC_REQ_PKT_BEGIN &&
+			    tx_entry->queued_ctrl.type <= RXR_DC_REQ_PKT_END) {
```

Comment:
This section can be written in a more general way that support all extra feature (not just delivery complete). Noticing we have a REQ_INF_LIST defined in ```rxr_pkt_type_req.c```, which for each extra feature we define a feature flag. We can used that flag here.

instead of
```
if (tx_entry->queued_ctrl.type >= RXR_DC_REQ_PKT_BEGIN &&
 			    tx_entry->queued_ctrl.type <= RXR_DC_REQ_PKT_END)
```
we do
```
if (tx_entry->queued_ctrl.type >= RXR_EXTRA_REQ_PKT_BEGIN &&
    tx_entry->queued_ctrl.type < RXR_EXTRA_REQ_PKT_END)
```

Reply:
queueing logic is no longer used now. resolving.